src/StreamJsonRpc/MessagePackFormatter.cs

```diff
@@ -1387,7 +1485,12 @@ internal JsonRpcRequestFormatter(MessagePackFormatter formatter)
                 this.formatter = formatter;
             }
 
-            public Protocol.JsonRpcRequest Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions options)
+            /// <summary>
+            /// Gets the raw msgpack encoding of the string "params".
+            /// </summary>
+            private static ReadOnlySpan<byte> ParamsPropertyNameBytes => new byte[] { 166, 112, 97, 114, 97, 109, 115 };
```

Comment:
I wonder if there would be value in moving all these fields to a static singleton that calculates these values via UTF8 encoder as a one time init cost. It would also make adding new strings in future easier.

Reply:
It's an interesting question. I certainly could initialize these at runtime. I'll leave this comment active to mull it over.

There is a method in MessagePack that can return these arrays (`CodeGenHelpers.GetEncodedStringBytes`). But I used https://kawanet.github.io/msgpack-lite/ which was even easier.
The harder values to get was the expected `ulong` result of `GetKey`. For that I had to write a little code to actually call the function with various strings to see what it would return. Alternatively folks can run mpc.exe on their project to get all this written out for them automatically, but StreamJsonRpc is a complex enough case I didn't want to try it. 😛

Reply:
At the very least I think we should check in that method to get ulong as a private utility method for future use.

Reply:
I pushed what amounts to a significant rewrite for readability and maintainability based on your feedback, @bertanaygun. Would you care to re-review?

Reply:
I had already approved before but these changes also look good to me. I agree that it is much easier to understand now and the details about string lenghts etc don't interfere with the serialization logic.